### PR TITLE
setup project structure and init code

### PR DESCRIPTION
**Pull request by thelonglqd**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["es2015", "stage-2"],
+  "plugins": []
+}
```
Comment (tungnt-580) [inline .babelrc]: Please add one more line here (https://github.com/airbnb/javascript#whitespace--newline-at-end)

File: models/users.model.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      .skip(+skip)
+      .limit(+limit)
+      .exec();
+  }
+};
+
+/**
+ * @typedef User
+ */
+export default mongoose.model('User', UserSchema);
```
Comment (tungnt-580) [inline models/users.model.js]: Please add one more line here (https://github.com/airbnb/javascript#whitespace--newline-at-end)

File: routes/users.route.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+router.route('/')
+  /** GET /api/users - Get list of users */
+  .get(userCtrl.list)
+
+  /** POST /api/users - Create new user */
+  // .post(validate(paramValidation.createUser), userCtrl.create);
+  .post(userCtrl.create);
+
+export default router;
```
Comment (tungnt-580) [inline routes/users.route.js]: Please add one more line here (https://github.com/airbnb/javascript#whitespace--newline-at-end)

File: .babelrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["es2015", "stage-2"],
```
Comment (tungnt-580) [inline .babelrc]: Dùng stage-3 đi đại ca. Có rest, spead operator

File: .gitignore [hunk 5]
```diff
@@ -57,3 +57,5 @@ typings/
 # dotenv environment variables file
 .env
 
+#package-lock.json
+package-lock.json
```
Comment (tungnt-580) [inline .gitignore]: https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5
TL;DR: Nên commit package-lock.json, mục đích chính là để các thành viên trong nhóm dev sử dụng cùng phiên bản của các dependencies

File: controllers/users.controller.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    password: req.body.password,
+    email: req.body.email
+  });
+
+  user.save()
+    .then(savedUser => res.json(savedUser))
+    .catch(e => next(e));
+}
+
+export default { load, get, create, update, list, remove };
```
Comment (tungnt-580) [inline controllers/users.controller.js]: Please add one more line here (https://github.com/airbnb/javascript#whitespace--newline-at-end)

**Comment by ngminhtrung** [comment]
LGTM (**L**ook **G**ood **T**o **M**e)
